scrapy/pipelines/images.py

```diff
@@ -126,16 +126,49 @@ def get_images(self, response, request, info):
         if width < self.min_width or height < self.min_height:
             raise ImageException("Image too small (%dx%d < %dx%d)" %
                                  (width, height, self.min_width, self.min_height))
+        
+        def _is_convert_image_overriden():
+            import inspect
+            if six.PY2:
+                convert_image_signature = inspect.getargspec(self.convert_image)
+            elif six.PY3:
+                convert_image_signature = inspect.getfullargspec(self.convert_image)
+            if 'response_body' not in convert_image_signature.args:
+                return True
+            return False
 
-        image, buf = self.convert_image(orig_image)
+        def _warn():
+            from scrapy.exceptions import ScrapyDeprecationWarning
+            import warnings
+            warnings.warn('ImagesPipeline.convert_image() method overriden in a incompatible way, '
+                          'overriden method does not accept response_body argument.',
+                          category=ScrapyDeprecationWarning, stacklevel=1)
+
+        convert_image_overriden = _is_convert_image_overriden()
+        if convert_image_overriden:
+            _warn()
+            image, buf = self.convert_image(orig_image)
+        else:
+            image, buf = self.convert_image(orig_image, response_body=BytesIO(response.body))
         yield path, image, buf
 
         for thumb_id, size in six.iteritems(self.thumbs):
             thumb_path = self.thumb_path(request, thumb_id, response=response, info=info)
-            thumb_image, thumb_buf = self.convert_image(image, size)
+            if convert_image_overriden:
+                _warn()
```

Comment:
I don’t think you need to log a warning each time you check the value of `convert_image_overriden`, logging the warning once should be enough.